Interop/Indexer/src/main/kotlin/org/jetbrains/kotlin/native/interop/indexer/J2ObjCParser.kt

```diff
@@ -0,0 +1,218 @@
+    )
+    return generatedClass
+  }
+
+  /**
+   * Creates a ObjCMethod out of method data and parameter names
+   *
+   * @param methodName Name of method
+   * @param methodDesc Descriptor of method
+   * @param access Integer that describes access level of this method
+   * @param paramNames List of parameter names of this method taken from MethodBuilder
+   */
+  private fun buildClassMethod(methodName: String, methodDesc: String, access: Int, paramNames: List<String>): ObjCMethod {
+    if (methodName == "<init>") {
+      return ObjCMethod(
+        selector = "init",
+        encoding = "[]",
+        parameters = listOf<Parameter>(), // TODO: Support constructor arguments
+        returnType = ObjCInstanceType(nullability = ObjCPointer.Nullability.Unspecified),
+        isVariadic = false,
+        isClass = false,
+        nsConsumesSelf = true,
+        nsReturnsRetained = true,
+        isOptional = false,
+        isInit = true,
+        isExplicitlyDesignatedInitializer = false)
+    } else {
+      val selector = StringBuilder(methodName)
+      val methodParameters = parseMethodParameters(methodDesc, paramNames)
+      val methodReturnType = parseMethodReturnType(methodDesc)
+      if (methodParameters.size >= 1) methodParameters.subList(1,methodParameters.size).forEach { selector.append(":" + it.name) }
+      return ObjCMethod(
+        selector = if (methodParameters.size > 1) "$selector:" else if (methodParameters.size == 1) "$methodName:" else methodName,
+        encoding = "[]", //TODO: Implement encoding properly
+        parameters = methodParameters,
+        returnType = methodReturnType,
+        isVariadic = false,
+        isClass = access == Opcodes.ACC_PUBLIC + Opcodes.ACC_STATIC, // TODO: Currently only handles Public instance and Public static methods, true when static, false when instance
+        nsConsumesSelf = false,
+        nsReturnsRetained = false,
+        isOptional = false,
+        isInit = false,
+        isExplicitlyDesignatedInitializer = false)
+    }
+  }
+
+  /**
+   * Parses an ASM method's parameters and returns a list of Kotlin parameters
+   *
+   * @param method ASM method descriptor
+   * @return A list of Kotlin parameters with associated types/names
+   */
+
+  private fun parseMethodParameters(methodDesc: String, paramNames: List<String>): List<Parameter> {
+    val methodParameters = mutableListOf<Parameter>()
+    val parameterTypes = getArgumentTypes(methodDesc)
+
+    for (i in 0 until parameterTypes.size) {
+      val paramName = paramNames.get(i)
+      when (parameterTypes.get(i).className) {
+        "boolean" -> methodParameters.add(Parameter(name = paramName, type = parseType(parameterTypes.get(i)), nsConsumed = false))
+        "byte" -> methodParameters.add(Parameter(name = paramName, type = parseType(parameterTypes.get(i)), nsConsumed = false))
```

Comment:
since I believe all these cases are the same, this can be written as:

"boolean", "byte", ..., "short" -> methodParamters.add(...)

Reply:
Done.